bayes/inference_problem.py

```diff
@@ -1,15 +1,42 @@
 import numpy as np
 from .parameters import ParameterList
 from .latent import LatentParameters
-from .noise import SingleNoise
+from .noise import SingleSensorNoise
 from collections import OrderedDict
-from .vb import variational_bayes, MVN, Gamma
+from .vb import MVN, Gamma, variational_bayes, VariationalBayesModelError
+from .jacobian import *
 
 
-class ModelError:
+class ModelErrorInterface:
+    def __init__(self):
+        self.parameter_list = ParameterList()
+
     def __call__(self):
+        """
+        Evaluate the model error based on `self.parameter_list` as a dict of
+        {some_key: numpy.array}.
+        """
         raise NotImplementedError("Override this!")
 
+    def jacobian(self, latent_names=None):
+        jac = dict()
+        latent_names = latent_names or self.parameter_list.names
+        for prm_name in latent_names:
+
+            try:
+                N = len(self.parameter_list[prm_name])
```

Comment:
The line with N was used to trigger the exception in case there is no `__len__`, N itself is unused, true. I'll move it somewhere else to not cause confusion here. 

For vector parameters, we have an additional loop over all entries of the vector. I could think of interpreting a scalar parameter as a vector parameter of length 1, but the access `p = prm[name]` vs `p = prm[name][row]` is really different. So far I thought that combining both cases would cause more confusion than the current state. Ideas welcome though!